File: frontend/src/app/modules/task/model/task.ts
import {User} from '../../user/model/user';
import {Project} from '../../project/model/project';

export class Task {
  id: number;
  createDate: number;
  description: string;
  dueDate: number;
  estimation: number;
  priority: string;
  status: string;
  ticketCode: string;
  updateDate: number;
  assignee: User;
  project: Project;
  reporter: User;

  costructor() {}
}

File: fapi/src/main/java/com/netcracker/lazarev/tms/model/Task.java
package com.netcracker.lazarev.tms.model;


import lombok.Data;

import javax.validation.constraints.NotEmpty;
import javax.validation.constraints.NotNull;

@Data
public class Task {
    private Long id;
    private Long createDate;
    @NotEmpty
    private String description;
    @NotNull
    private Long dueDate;
    @NotNull
    private Long estimation;
    @NotNull
    private String priority;
    @NotNull
    private String status;
    private String ticketCode;
    private Long updateDate;
    @NotNull
    private User assignee;
    @NotNull
    private Project project;
    private User reporter;
}
